plugins/thumb/thumb_mov.ml

```diff
@@ -29,123 +29,114 @@ module Make(CT : Theory.Core) = struct
       zf := is_zero (var rd);
     ]
 
-  let addi3 rd rn x = with_result rd @@ fun r -> [
-      r := var rn + const x;
+  let addi3 rd rn x = it_set rd (var rn + const x) @@ fun r -> [
       nf := msb (var r);
       zf := is_zero (var r);
       cf := carry_from_add (var r) (var rn);
       vf := overflow_from_add (var r) (var rn) (const x);
     ]
 
-  let addi8 rd x = with_result rd @@ fun r -> [
-      r := var rd + const x;
+  let addi8 rd x = it_set rd (var rd + const x) @@ fun r -> [
       nf := msb (var r);
       zf := is_zero (var r);
       cf := carry_from_add (var r) (var rd);
       vf := overflow_from_add (var r) (var rd) (const x);
     ]
 
-  let addrr rd rn rm = with_result rd @@ fun r -> [
-      r := var rn + var rm;
+  let addrr rd rn rm = it_set rd (var rn + var rm) @@ fun r -> [
       nf := msb (var r);
       zf := is_zero (var r);
       cf := carry_from_add (var r) (var rn);
       vf := overflow_from_add (var r) (var rn) (var rm);
     ]
 
-
-  let adcs rd rn rm = with_result rd @@ fun r -> [
-      r := var rn + var rm + CT.unsigned s32 (var zf);
+  let adcs rd rn rm =
+    it_set rd (var rn + var rm + CT.unsigned s32 (var zf)) @@ fun r -> [
```

Comment:
I think the flag to be added here should be `cf` instead of `zf`. Is that correct? The original code seems to have had this mistake.

Reply:
Good catch, thanks!